src/com/google/enterprise/adaptor/DocumentHandler.java

```diff
@@ -221,6 +243,10 @@ private boolean addressIsInSkipCertAddresses(InetAddress addr) {
   private boolean requestIsFromFullyTrustedClient(HttpExchange ex) {
     boolean trust;
     if (ex instanceof HttpsExchange) {
+      trust = requestIsFromSkipCertAddresses(ex);
+      if (trust) {
+        return true;
```

Comment:
This early return does more than skip the check. If requestIsFromSkipCertAddress the only thing we should do is skip the cert check.  We should skip ahead to the line pointed out in previous message. IMO.